src/game/Tactical/Soldier_Profile.cc

```diff
@@ -801,23 +802,12 @@ BOOLEAN UnRecruitEPC(ProfileID const pid)
 	p.ubMiscFlags &= ~PROFILE_MISC_FLAG_EPCACTIVE;
 
 	// update sector values to current
+	p.sSector = s->sSector;
 
-	// check to see if this person should disappear from the map after this
-	if ((pid == JOHN || pid == MARY) &&
-			s->sSector.x == 13            &&
-			s->sSector.y == MAP_ROW_B     &&
-			s->sSector.z == 0)
-	{
-		p.sSector = SGPSector();
-	}
-	else
-	{
-		p.sSector = s->sSector;
-	}
-
-	// how do we decide whether or not to set this?
-	p.fUseProfileInsertionInfo  = TRUE;
-	p.ubMiscFlags3             |= PROFILE_MISC_FLAG3_PERMANENT_INSERTION_CODE;
+	gMercProfiles[pid].ubMiscFlags3 |= PROFILE_MISC_FLAG3_PERMANENT_INSERTION_CODE;
+	gMercProfiles[pid].ubStrategicInsertionCode = INSERTION_CODE_GRIDNO;
+	gMercProfiles[pid].usStrategicInsertionData = s->sGridNo;
+	gMercProfiles[pid].fUseProfileInsertionInfo = TRUE;
```

Comment:
Why don't you use `p` like the rest of the code?

Reply:
Strange but my IDE indicated some of its members are inaccessible. Yet now it's all fine with it. It's probably my slow notebook's problem.